### PR TITLE
Add docs.nats.io examples to main

### PR DESCRIPTION
**Pull request by Jarema**
Moves the documentation examples from the `core-docs` and `jetstream-docs` branches into `examples/Example.NatsIODocs/` on main (52 new `Learn*.cs` files joining the 19 already here; additive only — csproj is auto-globbing and already in `NATS.Net.slnx`, so no project/solution changes).

Why: the new docs.nats.io build fetches these snippets at build time. On side branches they get zero CI; on main, `test_natsiodocs.yml` compiles AND executes them as xunit tests against a real nats-server on 3 OSes. Verified locally: `dotnet build` of the project passes with 0 warnings.

Note: please keep the docs branches until the docs repo's fetch config is switched to `main` (follow-up PR there).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: examples/Example.NatsIODocs/LearnCoreNatsPublishSubscribePublish.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var sub = await client.Connection.SubscribeCoreAsync<string>("orders.created");
+
+        // NATS-DOC-START
+        // Publish one order to the orders.created subject. Publishing is
+        // fire-and-forget: the call hands the message to the server and returns.
+        var order = """{"order_id":"ord_8w2k","customer":"acme-co","total_cents":4200,"ts":"2026-05-22T10:14:22Z"}""";
```
Comment (mtmk) [inline examples/Example.NatsIODocs/LearnCoreNatsPublishSubscribePublish.cs]: its more idiomatic for .net client to work with objects than json strings since ad-hoc json serialization is on by default.

Comment (Jarema) [reply examples/Example.NatsIODocs/LearnCoreNatsPublishSubscribePublish.cs]: Good point. switched the pub/sub example to publish a plain object and let the client's default JSON
serialization handle it (kept snake_case member names so the payload matches the other language tabs inthe docs). btw the same JSON-string pattern is in most of the other C# examples here.
wanna apply the object style across all of them, or keep it to the intro example?

**Review by mtmk — APPROVED** [review]
LGTM

File: examples/Example.NatsIODocs/LearnJetStreamAdvancedPublishingAsync.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+public class LearnJetStreamAdvancedPublishingAsync(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // The ORDERS stream captures every subject under `orders.`
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamAdvancedPublishingAsync.cs:22]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamPublishingSync.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public class LearnJetStreamPublishingSync(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // The ORDERS stream captures every subject under `orders.`
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamPublishingSync.cs:23]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamPublishingConfirmStored.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+public class LearnJetStreamPublishingConfirmStored(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // The ORDERS stream captures every subject under `orders.`
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamPublishingConfirmStored.cs:23]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamPublishingPubAck.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+public class LearnJetStreamPublishingPubAck(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // The ORDERS stream captures every subject under `orders.`
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamPublishingPubAck.cs:23]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamPublishingDedup.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+public class LearnJetStreamPublishingDedup(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // The ORDERS stream captures every subject under `orders.`
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamPublishingDedup.cs:23]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamMirrorsAndSourcesCreateMirror.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+public class LearnJetStreamMirrorsAndSourcesCreateMirror(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // The upstream the mirror follows must exist first
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamMirrorsAndSourcesCreateMirror.cs:31]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamMirrorsAndSourcesMirrorLag.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+public class LearnJetStreamMirrorsAndSourcesMirrorLag(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // Make sure the upstream and the mirror exist before we ask about lag
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamMirrorsAndSourcesMirrorLag.cs:31]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamMirrorsAndSourcesCreateSource.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+public class LearnJetStreamMirrorsAndSourcesCreateSource(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // Setup: the three regional streams ALL-ORDERS aggregates
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS-US", subjects: ["us.orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamMirrorsAndSourcesCreateSource.cs:17]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamYourFirstStreamCreate.cs [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // NATS-DOC-START
+        // Create the ORDERS stream, capturing every subject under `orders.`
+        var stream = await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamYourFirstStreamCreate.cs:18]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamYourFirstStreamInfo.cs [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+public class LearnJetStreamYourFirstStreamInfo(NatsServerFixture fixture, ITestOutputHelper output)
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        await using var client = new NatsClient(fixture.Server.Url);
+        var js = client.CreateJetStreamContext();
+
+        // Make sure the ORDERS stream exists before we ask about it
+        await js.CreateStreamAsync(new StreamConfig(name: "ORDERS", subjects: ["orders.>"]));
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamYourFirstStreamInfo.cs:17]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnJetStreamReadingBackRead.cs [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+        });
+
+        var read = 0;
+
+        // NATS-DOC-START
+        // Bind to the existing durable consumer
+        var consumer = await js.GetConsumerAsync("ORDERS", "billing");
+
+        // Read exactly what the stream is holding, no count assumed up front
+        long pending = (long)consumer.Info.NumPending;
```
[the remark this answers is not in the document]
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnJetStreamReadingBackRead.cs]: These are documentation examples and CI is green. We are intentionally not hardening them against cross-test edge cases such as shared-server config drift. Resolving.

File: examples/Example.NatsIODocs/LearnCoreNatsPublishSubscribePublish.cs (hunk 1 above)
Comment (mtmk) [reply examples/Example.NatsIODocs/LearnCoreNatsPublishSubscribePublish.cs]: Implemented. The C# docs examples now use records with the client's default JSON serialization, keeping snake_case member names so the payloads match the other language tabs. Resolving as done.
